### PR TITLE
Editorial: Tweak EnumerateObjectProperties informative definition

### PR DESCRIPTION
**Pull request by shvaikalesh**
1. Add missing semicolon.
1. Improve variable naming consistency (`protoName` --> `protoKey`, because `key`).
1. Early `continue` for `symbol`s (less indentation, more obvious what keys are skipped).
1. Use `const` instead of `let`.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -16175,16 +16175,16 @@
               for (let key of Reflect.ownKeys(obj)) {
                 if (typeof key === "string") {
                   let desc = Reflect.getOwnPropertyDescriptor(obj, key);
-                  if (desc && !visited.has(key)) {
+                  if (desc &amp;&amp; !visited.has(key)) {
```
Comment (claudepache) [inline spec.html]: Why? it just makes the source code less readable.

Comment (mathiasbynens) [inline spec.html]: Agreed. @shvaikalesh See https://mathiasbynens.be/notes/ambiguous-ampersands for more info on why this is perfectly valid.

File: spec.html [hunk 2]
```diff
@@ -16175,16 +16175,16 @@
               for (let key of Reflect.ownKeys(obj)) {
```
Comment (mathiasbynens) [inline spec.html]: Shouldn’t we use `const` over `let` throughout this code sample? None of the `let`s are reassigned.

File: spec.html (hunk 1 above)
Comment (shvaikalesh) [inline spec.html]: @mathiasbynens, thanks for the article. I dropped this commit and will make another PR to unescape all unambiguous ampersands in `spec.html`.

File: spec.html (hunk 2 above)
Comment (shvaikalesh) [inline spec.html]: It is mostly a matter of code style. Some might argue that `const` is for declarations that will never ever be reassigned, like number constants or `require`s. And `let`s are for any other declarations: this keeps diffs cleaner.

Lets make them `const`s to reflect how most people write ES201* these days.

File: spec.html [hunk 3]
```diff
@@ -16171,20 +16171,19 @@
...
-              let visited = new Set;
-              for (let key of Reflect.ownKeys(obj)) {
-                if (typeof key === "string") {
-                  let desc = Reflect.getOwnPropertyDescriptor(obj, key);
-                  if (desc && !visited.has(key)) {
-                    visited.add(key);
-                    if (desc.enumerable) yield key;
-                  }
+              const visited = new Set;
+              for (const key of Reflect.ownKeys(obj)) {
```
Comment (leobalter) [inline spec.html]: there's no tweak here. This change is not necessary at all and for consistency it is better to remain without change.

File: spec.html [hunk 4]
```diff
@@ -16171,20 +16171,19 @@
...
+                if (typeof key === "symbol") continue;
+                const desc = Reflect.getOwnPropertyDescriptor(obj, key);
+                if (desc && !visited.has(key)) {
+                  visited.add(key);
+                  if (desc.enumerable) yield key;
                 }
               }
-              let proto = Reflect.getPrototypeOf(obj)
+              const proto = Reflect.getPrototypeOf(obj);
```
Comment (leobalter) [inline spec.html]: this `;` is the only part I agree we should merge. The remaining are unnecessary as TC39 does not advise using const over let or against it.

File: spec.html (hunk 3 above)
Comment (shvaikalesh) [inline spec.html]: @leobalter, thanks for the review. Is it https://github.com/tc39/ecma262/pull/656/commits/46b569dc9f9ad621c2c97a022eae3d7e8603a3e3 or https://github.com/tc39/ecma262/pull/656/commits/f1883f50a6c7df79902f01c660486b889bdd7b07 that should be dropped?

Comment (leobalter) [inline spec.html]: I'm just part of TC39, @bterlson is the one able to answer that.

File: spec.html (hunk 4 above)
Comment (shvaikalesh) [inline spec.html]: Any JavaScript code contains opinionated parts: this function will be just fine without semicolons and strict equals. Does TC39 advise against `==` and ASI too?

File: spec.html (hunk 3 above)
Comment (ljharb) [inline spec.html]: I think it's ok to change it to use const, since it's never reassigned. I'd also want to see `new Set()` while we're at it :-)

File: spec.html (hunk 4 above)
Comment (ljharb) [inline spec.html]: TC39 isn't advising one over the other in either case. Since both are equivalent, and the community tends to prefer const here, I thi k that's the better choice.

File: spec.html (hunk 3 above)
Comment (bterlson) [inline spec.html]: I am fine with this change. I don't see why this change isn't consistent though - @leobalter can you clarify?

Comment (leobalter) [inline spec.html]: I prefer this example being consistent (the same) along the new releases.

+ replacing let w/ const will make it valid to replace const w/ let in any near future as we don't enforce any particular use.

Comment (bterlson) [inline spec.html]: We do enforce a particular use in the spec at least - whatever I feel like 😈

My feelings may change (I have in fact become far less likely to use const) but for now I think this helps and I promise I won't change it back to let any time soon.

Comment (shvaikalesh) [inline spec.html]: @ljharb addressed in https://github.com/tc39/ecma262/pull/656/commits/aff0edd611d4c460225a681bf17455f2eaf3fbe7
